datastore/blob/azure_storage_api.py

```diff
@@ -0,0 +1,266 @@
+import json
+import uuid
+import hashlib
+import os
+import datetime
+from azure.storage.blob import BlobServiceClient
+
+from custom_exceptions import (
```

Comment:
exceptions should be per module. also ensure that they share a base class AzureStorageApiError so the caller can catch all exceptions from this module in a single except.